runtime/compiler/env/VMJ9.h

```diff
@@ -1140,8 +1140,29 @@ class TR_J9VMBase : public TR_FrontEnd
    virtual bool      isJavaLangObject(TR_OpaqueClassBlock *clazz);
    virtual int32_t   getStringLength(uintptr_t objectPointer);
    virtual uint16_t  getStringCharacter(uintptr_t objectPointer, int32_t index);
-   virtual intptr_t getStringUTF8Length(uintptr_t objectPointer);
-   virtual char     *getStringUTF8      (uintptr_t objectPointer, char *buffer, intptr_t bufferSize);
+
+   /**
+    * \brief Returns the number of UTF-8 encoded bytes needed to represent a Java String object.
+    *        The number of bytes needed to UTF-8 encode the String is representable as
+    *        a \c uintptr_t, in general, but this method returns a length of type \c int32_t.
```

Comment:
`uintptr_t` might not be enough on 32-bit, so in general we'd need `uint64_t`, which I think is why you made that the return type of `getStringUTF8UnabbreviatedLength()`

Reply:
Oops!  Thanks for catching that.

Reply:
Fixed in commit a4974d4242c9205873e6ad6498cb9c724c536cde